plugins/inputs/prometheus/README.md

```diff
@@ -28,8 +28,17 @@ in Prometheus format.
   
   ## Kubernetes config file to create client from.
   # kube_config = "/path/to/kubernetes.config"
+
+  ## Explicit scrape configuration for pods matched by label, field and namespace selectors
+  # Values can't be overridden by the `prometheus.io/` pod annotations. Not specified or
+  # enabled = false, then scraping config is entirely governed by pod annotations.
+  # [inputs.prometheus.scrape_config]
```

Comment:
In TOML order matters, so these types of definitions need to be at the end of a config as any key, pair after this header will be considered part of `scape-config`.

To keep this from getting confusing, I would prefer we did something like the following *after* the `monitor_kubernetes_pods` setting:

```toml
## Method to determine pods to monitor
## Set to "annotations" only pods matching the above `prometheus.io/scrape`
## annotation and other settings are monitored. Set to "options" will use the
## configured option set below.
# monitor_kubernetes_pods_method = ["annotations"]
# monitor_kubernetes_pods_scheme = "http"
# monitor_kubernetes_pods_path = "/metrics"
# monitor_kubernetes_pods_port = 9102
```

Does that make this a bit more straightforward as well? What do you think about that naming?

Reply:
`monitor_kubernetes_pods_method = "options"` should continue to use annotations if they are set, right?

Reply:
Ah - maybe I misunderstood your intention. You still want to scrape those with `prometheus.io/scrape`, only ommit the additional scheme, path, and port annotations?

If so, then sounds like we can remove the `monitor_kubernetes_pods_method` option, and just have the scheme, path, and port options set to their defaults.

Reply:
You understood correctly, I want to be able to scrape even if pod doesn't have  any annotations. However, I'd like pod annotations  to have higher priority, so pod can override scheme,path, port or even disable scraping with   `prometheus.io/scrape: false` annotaion

Reply:
Do these descriptions match your intention better:

* annotations: the default, and the current behavior of only scraping pods with the `prometheus.io/scape` and other annotations
* fallback: attempts to scrape any pod it finds, looks at the annotations first, and falls back to values found in settings?

Reply:
your described behaviour is exactly what I had in mind. "fallback" name though is a bit misleading IMHO, as it implies logic to execute if something didn't work, but in reality it is just another way of  configuring scraping.

Reply:
Feel free to come up with a different name :) not sold on that either. Combo? All? Both?

Reply:
"config" ?

Reply:
My concern with config is it makes it sound like it would only scrape items based on the config, versus your proposal to do both annotations + config.

I could see config being a third option, where you only scrape based on the specified config options.

Thoughts?

Reply:
> My concern with config is it makes it sound like it would only scrape items based on the config, versus your proposal to do both annotations + config.

The way I see it  that "config"  is capable of scraping entirely based on config, it just leaving pod option to alter scraping behavior with annotation, but primary driver is still a config.

> I could see config being a third option, where you only scrape based on the specified config options.

It not a third option, but a variation of config option, but disabling annotations overrides. If we add it, it should probably enabled with `monitor_kubernetes_pods_ignore_annotations` or `monitor_kubernetes_pods_force` flag.

Reply:
We avoid boolean flags wherever possible. That is why I was trying to think through what that third option would be using this method variable. Otherwise you get into even more combinations that confuse users when you have different methods + booleans that only affect certain methods.

Reply:
OK, give me 2 (3) method names, I'll implement them as discussed. In the absence of better ideas I am happy take your lead  as my goal is to get feature in :)

Reply:
How about:

- annotations
- settings
- annotations+settings

Does that make it even more clear?

Reply:
@powersj , I've updated PR with implementation of your proposed config options